### PR TITLE
Shipping price showing different in printed pdf

### LINKED ISSUE #38608: [Issue] Shipping price showing diffrent in printed pdf
This issue is automatically created based on existing pull request: magento/magento2#38595: Shipping price showing diffrent in printed pdf

---------

### Description (*)
In Stores -> Configuration -> Sales -> Tax -> In the "Display Shipping Price" tab, we can specify whether to show shipping prices including or excluding tax. As seen in screenshot 1, I've set the "Display Shipping Price" option to "Including Tax." Consequently, in screenshot 2, the shipping price is displayed including tax on the sales-order_invoice-view page. However, when we print the order in PDF format from this page, the shipping price is shown as excluding tax, which is incorrect. It should be displayed as per tax configuration and similar to the sales-order_invoice-view page because we are printing the page. You can view the printed PDF in screenshot 3.

**Screenshot-1:**
image

**Screenshot-2:**
image

**Screenshot-3:**
image

### Manual testing scenarios (*)
1:  Navigate to  Admin->Store->Sales->Tax and Select the option as shown in screenshot 4 &5
**Screenshot-4:**
image

**Screenshot 5:**
image

2: create a one tax rule you can see the screenshot 6 i have created one tax rule with 25% tax rate
**Screenshot-6:**
image

3:Now order one product from the frontend 
4:Now navigate to admin side and create a invoice of that order 
5:Now, navigate to Sales -> Invoice and view the invoice you created. 
6:now print the order from invoice page as you can see the diffrence of the shipping price in the sales-order_invoice-view page and in the pdf as depricated in screenshot 2 & 3.

### Expected result (*)
The shipping charges should be shown the same way they are displayed on the sales-order_invoice-view page since we are printing the page.

### Actual result (*)
The shipping charges are displayed differently on the PDF compared to the sales order invoice view page

### Aditional information 
 The file public_html/app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php is responsible for the content seen in the printed PDF. The logic for displaying shipping charges is found in lines 607 to 610 of this file. However, there is no logic implemented to show shipping charges according to the configuration settings. On the other hand, in public_html/app/code/Magento/Sales/view/adminhtml/templates/order/invoice/create/form.phtml, the logic is implemented to fetch the configuration and then display the shipping charges accordingly on the sales-order_invoice-view page. You can see the logic of form.phtml in Screenshot 7, where shipping charges are displayed according to the configuration. However, in AbstractPdf.php, there is no such logic implemented, and only the shipping price excluding tax is shown, as depicted in Screenshot 8.
 
 **Screenshot 7:**
image

**Screenshot-8:**
image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by DhruvBasan**
### Description (*)
In Stores -> Configuration -> Sales -> Tax -> In the "Display Shipping Price" tab, we can specify whether to show shipping prices including or excluding tax. As seen in screenshot 1, I've set the "Display Shipping Price" option to "Including Tax." Consequently, in screenshot 2, the shipping price is displayed including tax on the sales-order_invoice-view page. However, when we print the order in PDF format from this page, the shipping price is shown as excluding tax, which is incorrect. It should be displayed as per tax configuration and similar to the sales-order_invoice-view page because we are printing the page. You can view the printed PDF in screenshot 3.

**Screenshot-1:**
image

**Screenshot-2:**
image

**Screenshot-3:**
image

### Manual testing scenarios (*)
1:  Navigate to  Admin->Store->Sales->Tax and Select the option as shown in screenshot 4 &5
**Screenshot-4:**
image

**Screenshot 5:**
image

2: create a one tax rule you can see the screenshot 6 i have created one tax rule with 25% tax rate
**Screenshot-6:**
image

3:Now order one product from the frontend 
4:Now navigate to admin side and create a invoice of that order 
5:Now, navigate to Sales -> Invoice and view the invoice you created. 
6:now print the order from invoice page as you can see the diffrence of the shipping price in the sales-order_invoice-view page and in the pdf as depricated in screenshot 2 & 3.

### Expected result (*)
The shipping charges should be shown the same way they are displayed on the sales-order_invoice-view page since we are printing the page.

### Actual result (*)
The shipping charges are displayed differently on the PDF compared to the sales order invoice view page

### Aditional information 
 The file public_html/app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php is responsible for the content seen in the printed PDF. The logic for displaying shipping charges is found in lines 607 to 610 of this file. However, there is no logic implemented to show shipping charges according to the configuration settings. On the other hand, in public_html/app/code/Magento/Sales/view/adminhtml/templates/order/invoice/create/form.phtml, the logic is implemented to fetch the configuration and then display the shipping charges accordingly on the sales-order_invoice-view page. You can see the logic of form.phtml in Screenshot 7, where shipping charges are displayed according to the configuration. However, in AbstractPdf.php, there is no such logic implemented, and only the shipping price excluding tax is shown, as depicted in Screenshot 8.
 
 **Screenshot 7:**
image

**Screenshot-8:**
image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#38608: Shipping price showing diffrent in printed pdf

### CODE REVIEW

**Comment by DhruvBasan** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento create issue

**Comment by engcom-Hotel** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php [hunk 1]
```diff
@@ -131,6 +132,11 @@ abstract public function getPdf();
      */
     protected $addressRenderer;
 
+    /**
+     * @var Magento\Tax\Helper\Data
+     */
+    protected $taxHelper;
```
Comment (engcom-Hotel) [inline app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php]: Make this variable as `private`

File: app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php [hunk 2]
```diff
@@ -169,6 +176,7 @@ public function __construct(
         \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate,
         \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation,
         \Magento\Sales\Model\Order\Address\Renderer $addressRenderer,
+        TaxHelper $taxHelper,
```
Comment (engcom-Hotel) [inline app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php]: Make this `null` by default to maintain backward compatibility.

File: app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php [hunk 3]
```diff
@@ -185,6 +193,7 @@ public function __construct(
         $this->_pdfTotalFactory = $pdfTotalFactory;
         $this->_pdfItemsFactory = $pdfItemsFactory;
         $this->inlineTranslation = $inlineTranslation;
+        $this->taxHelper = $taxHelper;
```
Comment (engcom-Hotel) [inline app/code/Magento/Sales/Model/Order/Pdf/AbstractPdf.php]: Use the ObjectManager to initialize this variable.

File: app/code/Magento/Sales/Model/Order/Pdf/Creditmemo.php [hunk 4]
```diff
@@ -54,6 +55,7 @@ public function __construct(
         \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate,
         \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation,
         \Magento\Sales\Model\Order\Address\Renderer $addressRenderer,
+        \Magento\Tax\Helper\Data $taxHelper,
```
Comment (engcom-Hotel) [inline app/code/Magento/Sales/Model/Order/Pdf/Creditmemo.php]: Make this `null` by default

File: app/code/Magento/Sales/Model/Order/Pdf/Creditmemo.php [hunk 5]
```diff
@@ -71,6 +73,7 @@ public function __construct(
             $localeDate,
             $inlineTranslation,
             $addressRenderer,
+            $taxHelper,
```
Comment (engcom-Hotel) [inline app/code/Magento/Sales/Model/Order/Pdf/Creditmemo.php]: initialize $taxHelper with ObjectManager

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @DhruvBasan,

Thanks for the contribution!

Please check the below review comments, for Backward compatibility, please mark the newly added parameters to null in constructor and  use ObjectManager to initialize those.

Also please check the failed tests, it seems they are failing due to code changes.

Thanks

**Comment by engcom-Charlie** [comment]
Hi @DhruvBasan,

Thank you for your contribution!

To proceed this PR further, we need green build. Can you please have a look into Functional CE, Static, Unit, Integration, SVC test failures.


Thank you!

**Comment by engcom-Charlie** [comment]
Hi @DhruvBasan,

Thank you for your contribution!

Did you get time to work on the changes.

Thank you!

**Comment by DhruvBasan** [comment]
@engcom-Hotel  @engcom-Charlie I have made the changes to the files as per your suggestions. Kindly review them.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Thanks @DhruvBasan, for quick workaround, but it seems the failed tests are due to code changes. I Request you to please fix them.

Thanks

**Comment by engcom-Charlie** [comment]
Hi @DhruvBasan,

Thank you for your contribution!

Did you get time to work on the build failures.

Thank you!

**Comment by engcom-Bravo** [comment]
Hello,

Internal team has started to work on it

Thanks.

**Comment by engcom-Bravo** [comment]
Hello,

As I can see this issue got fixed in the scope of the internal Jira ticket AC-11798 by the internal team
Related commits:https://github.com/search?q=repo%3Amagento%2Fmagento2+AC-11798&type=commits

Based on the Jira ticket, the target version is 2.4.8-beta1.

Thanks

**Pull request closed without merging**